clash-lib/src/Clash/Core/Literal.hs

```diff
@@ -42,8 +43,8 @@ data Literal
   | Int64Literal    !Integer
   | Word64Literal   !Integer
   | StringLiteral   !String
-  | FloatLiteral    !Float
-  | DoubleLiteral   !Double
+  | FloatLiteral    !Word32
+  | DoubleLiteral   !Word64
```

Comment:
I think you should add a note about why we use `Word{32,64}` here instead of (1) `Rational` and (2) `Float`/`Double`.
Notes typically look something like this in source:

```haskell
{-
Note [name of note]

This is a note. It notes something.
-}
```

Reply:
Ah, right, that line of source code only triggers on old versions of GHC, that's why it didn't show up for me.

I'll add the code comment tomorrow-ish.